src/server/api/routers/scoreboard.ts

```diff
@@ -0,0 +1,102 @@
+// types.ts
+type Challenge = {
+  roundId: number;
+  points: number;
+  teamId: string;
+};
+
+type TeamScores = {
+  teamId: string;
+  teamName: string;
+  rounds: Record<
+    number,
+    {
+      challengeA: number;
+      challengeB: number;
+      challengeC: number;
+    }
+  >;
+  total: number;
+};
+
+// server/api/routers/scores.ts
+import { z } from "zod";
+import { createTRPCRouter, publicProcedure } from "../trpc";
+
+export const scoreboardRouter = createTRPCRouter({
+  getScoreboard: publicProcedure.query(async ({ ctx }) => {
+    // Fetch all teams
+    const teams = await ctx.db.team.findMany({
+      select: {
+        id: true,
+        name: true,
+      },
+    });
+
+    // Fetch all challenges
+    const [challengesA, challengesB, challengesC] = await Promise.all([
+      ctx.db.challengeA.findMany(),
+      ctx.db.challengeB.findMany(),
+      ctx.db.challengeC.findMany(),
+    ]);
+
+    // Process scores for each team
+    const teamScores: TeamScores[] = teams.map((team) => {
+      const scores: TeamScores = {
+        teamId: team.id,
+        teamName: team.name,
+        rounds: {},
+        total: 0,
+      };
+
+      // Initialize rounds data
+      [1, 2, 3].forEach((roundId) => {
```

Comment:
are we 100% sure about this values? If so can we enum and add it in the front-end?